### DISCUSSION TITLE
https://www.spsanderson.com/steveondata/posts/2024-10-04/

### DISCUSSION THREAD

**Discussion by giscus** [spsanderson/steveondata#11] [Comments]
# https://www.spsanderson.com/steveondata/posts/2024-10-04/

https://www.spsanderson.com/steveondata/posts/2024-10-04/

**Comment by matt-kita** [comment]
Hello.
Unfortunately, `compgen -c` only provides commands "runnable" from locations included in your `$PATH` (executables, including scripts [not only binaries], and does not account for duplicates [e.g. in some systems `lsmod` is present in both /bin and /sbin]).
If you actually wanted to see ALL commands (available to Bash, including shell builtin commands, shell keywords, defined aliases and functions) you'd most likely need:
`compgen -abck -A function`

**Reply by spsanderson** [reply]
Thank you for the clarification! Much appreciated.
